### DISCUSSION TITLE
Websocket connection with parameters

### DISCUSSION THREAD

**Discussion by IvDmNe** [foxglove/community#700] [Q&A] [answered]
Hello! I am using foxglove websocket to visualize my protobuf data in Studio. Exploring the last C++ (1.0.0) version, I noticed that support for parameters has been added. And I have 2 questions:
1. Why you decided to use standalone asio library instead of included in boost? In my stack I have RTPS that uses system asio. And when in one program I use some RTPS functions (didn't dive into it) and after it try to create foxglove server, inside asio a segfault is thrown when trying to access some uninitialized fields. Fortunately, I was able to change from standalone to boost version, applying a patch to websocket_server.hpp and websocket_logging.hpp. But I don't know, maybe there were some problems with boost version that make foxglove team to use standalone? 
2. How can I use panel Parameters in Studio with custom data? In [docs](https://foxglove.dev/docs/studio/panels/parameters) it is said that we can connect to custom data source. I created parameterRequestHandler and parameterSubscriptionHandler for my server, but the callbacks are never entered when at the same time I open Parameters panel in Studio.

**Comment by jtbandes** [comment, marked as answer]
For 2, did you include the [parameters capability](https://github.com/foxglove/ws-protocol/blob/main/cpp/foxglove-websocket/include/foxglove/websocket/common.hpp#L14) in your server config?

**Reply by IvDmNe** [reply]
Oh, No, I didn't. After adding this in config, parameters are correctly readed by Parameters panel. Thank you. Is there anywhere a documentation or guides on websocket API, except of example? Anyway, URDF model in 3D panel still cannot be automatically readed from parameters. I run websocket server on a remote PC. In URL I put a link to my local file, is it correct? The same link is valid when I manually add URDF model and set a URL.. Beside /robot_description (parameter) topic there is an error: "Failed to parse URDF: Failed to parse 48 byte URDF: Error: No robot found in URDF". I tried to pass char*, std::string and string literals in parameter constructor, but it still fails to parse it.

**Reply by jtbandes** [reply]
No, the parameter should be the whole contents of the urdf, not just a url.

**Reply by IvDmNe** [reply]
Yes, my mistake. Now my model is loaded and visualized correctly. Thank you for help!
